src/glue.c

```diff
@@ -214,6 +217,135 @@ PyObject *mg_path_to_py_path(const mg_path *path) {
+  PyDateTime_IMPORT;
+  SCOPED_CLEANUP PyObject *unix_epoch =
+      make_py_datetime(1970, 1, 1, 0, 0, 0, 0);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(unix_epoch, NULL)
+  SCOPED_CLEANUP PyObject *numerator =
+      PyLong_FromLong(mg_local_time_nanoseconds(lt));
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(numerator, NULL)
+  SCOPED_CLEANUP PyObject *denominator = PyLong_FromLong(1000000000);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(denominator, NULL)
+  SCOPED_CLEANUP PyObject *div_result =
+      PyNumber_TrueDivide(numerator, denominator);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(div_result, NULL)
+  SCOPED_CLEANUP PyObject *method_name =
+      PyUnicode_FromString("utcfromtimestamp");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(method_name, NULL)
+  SCOPED_CLEANUP PyObject *result =
+      PyObject_CallMethodObjArgs(unix_epoch, method_name, div_result, NULL);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(result, NULL);
+  SCOPED_CLEANUP PyObject *h = PyObject_GetAttrString(result, "hour");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(h, NULL)
+  SCOPED_CLEANUP PyObject *m = PyObject_GetAttrString(result, "minute");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(m, NULL)
+  SCOPED_CLEANUP PyObject *s = PyObject_GetAttrString(result, "second");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(s, NULL)
+  SCOPED_CLEANUP PyObject *mi = PyObject_GetAttrString(result, "microsecond");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(mi, NULL)
+  return make_py_time(PyLong_AsLong(h), PyLong_AsLong(m), PyLong_AsLong(s),
+                      PyLong_AsLong(mi));
+}
+
+PyObject *mg_local_date_time_to_py_datetime(const mg_local_date_time *ldt) {
+  PyDateTime_IMPORT;
+  SCOPED_CLEANUP PyObject *unix_epoch =
+      PyDateTime_FromDateAndTime(1970, 1, 1, 0, 0, 0, 0);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(unix_epoch, NULL)
+  SCOPED_CLEANUP PyObject *seconds =
+      PyLong_FromLong(mg_local_date_time_seconds(ldt));
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(seconds, NULL)
+  SCOPED_CLEANUP PyObject *method_name =
+      PyUnicode_FromString("utcfromtimestamp");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(method_name, NULL)
+  SCOPED_CLEANUP PyObject *result =
+      PyObject_CallMethodObjArgs(unix_epoch, method_name, seconds, NULL);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(result, NULL)
+  SCOPED_CLEANUP PyObject *y = PyObject_GetAttrString(result, "year");
+  SCOPED_CLEANUP PyObject *mo = PyObject_GetAttrString(result, "month");
+  SCOPED_CLEANUP PyObject *d = PyObject_GetAttrString(result, "day");
+  SCOPED_CLEANUP PyObject *h = PyObject_GetAttrString(result, "hour");
+  SCOPED_CLEANUP PyObject *m = PyObject_GetAttrString(result, "minute");
+  SCOPED_CLEANUP PyObject *s = PyObject_GetAttrString(result, "second");
+  int64_t nanos = mg_local_date_time_nanoseconds(ldt);
+  return make_py_datetime(PyLong_AsLong(y), PyLong_AsLong(mo), PyLong_AsLong(d),
+                          PyLong_AsLong(h), PyLong_AsLong(m), PyLong_AsLong(s),
+                          (nanos / 1000));
```

Comment:
Isn't `result` already a `datetime` object? I think there is no need to get the member of it and then reassemble it in `make_py_datetime`.

Reply:
Good point but there is a catch. `datetime` objects are immutable so you can't call `SetAttrString()` to set the `microseconds` field. Hence you either need to call `[replace](https://docs.python.org/3/library/datetime.html#datetime.datetime.replace)` or `make_py_datetime`. I prefer the second because I already have this utility function and I don't really like calling python methods from C.

Reply:
So here we only use python to convert the seconds to the appropriate values. Then I think this is fine.